test/e2e/storage/drivers/csi.go

```diff
@@ -608,17 +608,18 @@ func waitForCSIDriverRegistrationOnAllNodes(driverName string, cs clientset.Inte
 		return err
 	}
 	for _, node := range nodes.Items {
-		if err := waitForCSIDriverRegistrationOnNode(node.Name, driverName, cs); err != nil {
+		if err := WaitForCSIDriverRegistrationOnNode(node.Name, driverName, cs, 5*time.Minute); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func waitForCSIDriverRegistrationOnNode(nodeName string, driverName string, cs clientset.Interface) error {
-	const csiNodeRegisterTimeout = 5 * time.Minute
+// WaitForCSIDriverRegistrationOnNode waits for the CSINode object generated by the node-registrar on a certain node
+func WaitForCSIDriverRegistrationOnNode(nodeName string, driverName string, cs clientset.Interface, timeout time.Duration) error {
+	framework.Logf(fmt.Sprintf("waiting up to %v for CSIDriver %v to register on node %v", timeout, driverName, nodeName))
 
-	waitErr := wait.PollImmediate(10*time.Second, csiNodeRegisterTimeout, func() (bool, error) {
+	waitErr := wait.PollImmediate(10*time.Second, timeout, func() (bool, error) {
```

Comment:
not introduced this PR, but IMO we should use exponential backoff for checking CSINode registration. That might be less expensive on api-server and may cause these functions to return sooner in happy path. I am thinking something like - "initial: 2, factor: 1.4, count: 12"

Reply:
Done